Scripts/Core/EditorVR.Menus.cs

```diff
@@ -382,7 +382,21 @@ internal static bool IsValidHover(MultipleRayInputModule.RaycastSource source)
 
 				var eventData = source.eventData;
 				var rayOrigin = eventData.rayOrigin;
-				var deviceData = evr.m_DeviceData.FirstOrDefault(dd => dd.rayOrigin == rayOrigin);
+
+				DeviceData deviceData = null;
+				var deviceDataList = evr.m_DeviceData;
+				var dataCounter = 0;
+				while (dataCounter < deviceDataList.Count)
+				{
+					var currentDevice = deviceDataList[dataCounter];
+					if (currentDevice.rayOrigin == rayOrigin)
+					{
+						deviceData = currentDevice;
+						break;
+					}
+					dataCounter++;
+				}
```

Comment:
Why not just a for/foreach loop here? AFAIK, as of 5.6 foreach on lists and dictionaries no longer allocate. I think it was just the use of LINQ that got us here.

Reply:
That sounds right - I'll test & confirm

Reply:
correct - re-foreach'd